advancedlog/logger.go

```diff
@@ -0,0 +1,52 @@
+package logger
+
+import "os"
+
+const (
+	// JSONFormat ...
+	JSONFormat = "json"
+)
+
+// Logger ...
+type Logger interface {
+	Debug(args ...interface{})
+	Debugf(format string, args ...interface{})
+	Info(args ...interface{})
+	Infof(format string, args ...interface{})
+	Done(args ...interface{})
+	Donef(format string, args ...interface{})
+	Warn(args ...interface{})
+	Warnf(format string, args ...interface{})
+	Error(args ...interface{})
+	Errorf(format string, args ...interface{})
+	Fatal(args ...interface{})
+	Fatalf(format string, args ...interface{})
+	Print(args ...interface{})
+	Printf(format string, args ...interface{})
+	Println(args ...interface{})
+	TPrintf(format string, args ...interface{})
+	EnableDebugLog(enable bool)
+	IsDebugLogEnabled() bool
+}
+
+// SimplifiedLogger ...
+type SimplifiedLogger interface {
+	EnableDebugLog(enabled bool)
+	IsDebugLogEnabled() bool
+	LogMessage(producer Producer, level Level, message string)
+}
+
+// DefaultLogger ...
+var DefaultLogger = NewMainLogger()
+
+// SetupLogger ...
+func SetupLogger(outputFormat string) {
```

Comment:
IMO we could use a dedicated type for the output format:

```
type OutputFormat string

const (
	// JSONFormat ...
	JSONFormat OutputFormat = "json"
)

...

func SetupLogger(outputFormat OutputFormat) {
...
```

Reply:
It seems current you can configure two things on the logger:
- outputFormat
- whether debug mode is enabled

I think they should have similar function names, like:
- SetEnableDebugLog
- SetOutputFormat